planner/core/logical_plan_builder.go

```diff
@@ -2282,6 +2307,10 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint, nodeType u
 		switch hint.HintName.L {
 		case TiDBMergeJoin, HintSMJ:
 			sortMergeTables = append(sortMergeTables, tableNames2HintTableInfo(b.ctx, hint.Tables, b.hintProcessor, nodeType, currentLevel)...)
+		case TiDBBroadCastJoin, HintBCJ:
```

Comment:
what if there are conflicts among hints? for example, storage_type should include `tiflash`, then bcj can be used.

Reply:
If conflicts happen, like storage_type is tikv and bcj is applied, it will report an error like 'cannot find proper physical plan'